canvas/models/models.py

```diff
@@ -132,6 +132,14 @@ def total_tokens_received(self):
                 tokens += 0
             else:
                 tokens += event.tokens_received(team)
+
+        from course.models.models import UserQuestionJunction
+
+        practiced_uqjs = UserQuestionJunction.objects.filter(user=self.user, is_solved=True, question__event=None)
```

Comment:
The filters are incorrect. You dont need is solved as there might be partially solved too. You need question course = None and question status to be created and is verified to be true. Basically the same filters that i told you in one of the other PRS.

Reply:
Also try to avoid local imports. You can probably move the to the service folder of the course app.